### PR TITLE
Updated docs, cleaned things up, DRY's up function format stuff.

### PR DESCRIPTION
**Pull request by treeder**
* Made the Getting started list of docs better for walking through things
* Removed writing.md, in favor of function-format.md, they were essentially the same thing
* Moved everything developer related into developers dir

### CODE REVIEW

File: docs/developers/function-file.md [hunk 1]
```diff
@@ -2,19 +2,13 @@
 
 Functions files are used to assist fn to help you when creating functions.
 
-The files can be named as:
-
-- func.yaml
-- func.json
-
```
Comment (skinowski) [inline docs/developers/function-file.md]: intentional change to remove this?

File: docs/developers/function-format.md [hunk 2]
```diff
@@ -174,7 +144,79 @@ Cons:
...
+* Requires a parsing library or fair amount of code to parse headers properly
+* Double parsing - headers + body (if body is to be parsed, such as json)
+
 ## Logging
 
-Standard error is reserved for logging, like it was meant to be. Anything you output to STDERR will show up in the logs. And if you use a log
-collector like logspout, you can collect those logs in a central location. See [logging](logging.md).
+Standard out is where you should write response data for synchronous functions. Standard error
+is where you should write for logging, as [it was intended](http://www.jstorimer.com/blogs/workingwithcode/7766119-when-to-use-stderr-instead-of-stdout).
+And if you use a log collector like logspout, you can collect those logs in a central location. See [logging](logging.md).
```
Comment (skinowski) [inline docs/developers/function-format.md]: operating/logging.md?

File: docs/README.md [hunk 3]
```diff
@@ -50,8 +49,7 @@ If you are operating Fn, this section is for you.
 
 * [Extending Fn](operating/extending.md)
 * [Kubernetes Helm Chart for Fn](https://github.com/fnproject/fn-helm/)
-* [Setting up development environment with Docker compose](./operating/compose.md)
-* [OpenStack Triggers](operating/triggers.md)
```
Comment (skinowski) [inline docs/README.md]: intentional to remove triggers link?

File: docs/README.md [hunk 4]
```diff
@@ -7,30 +7,29 @@ If you are a developer using Fn through the API, this section is for you.
...
-* [API Reference](http://petstore.swagger.io/?url=https://raw.githubusercontent.com/fnproject/fn/master/docs/swagger.yml)
+* [Function file (func.yaml)](developers/function-file.md)
+* [FDKs](developers/fdks.md)
+* [Config Vars for Functions](developers/configs.md)
+* [Hot functions](developers/hot-functions.md)
+* [Testing functions](developers/testing.md)
+* [Applications - groups of functions](developers/apps.md)
+* [Async functions](developers/async.md)
 * [FAQ](faq.md)
-* [Object Model](developers/model.md)
```
Comment (skinowski) [inline docs/README.md]: intentional to remove object model?

Comment (treeder) [reply docs/README.md]: ya, don't think it's that useful and we don't keep it up to date. Can get same info from swagger.

File: docs/README.md (hunk 3 above)
Comment (treeder) [reply docs/README.md]: Ya, old stuff.

File: docs/developers/function-file.md (hunk 1 above)
Comment (treeder) [reply docs/developers/function-file.md]: We never talk about the json format, so might as well not promote it.

**Comment by michael-w-williams** [comment]
I think there might be some extra '+' symbols in function-file.md. Also, for bullet lists, make sure all items are consistent within a list. So all items end with a period, or all items do not have a period, for that list.

Otherwise it looks good to me.

**Pull request merged**